### PR TITLE
WL-3944 Show nested section structure  

### PR DESCRIPTION
**Pull request by nicholaswilson100**

### CODE REVIEW

**Comment by nicholaswilson100** [comment]
Linked to https://github.com/ox-it/wl-reference/pull/101

File: citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java [hunk 1]
```diff
@@ -312,254 +309,39 @@ protected void handleViewRequest(HttpServletRequest req, HttpServletResponse res
...
-    			out.println("\t\t<div class=\"itemAction\">");
-    			out.println("\t\t\t<a href=\"#\" onclick=\"showAllDetails( '" + rb.getString("link.hide.results") + "' ); return false;\">" + rb.getString("link.show.readonly") + "</a> |" );
-    			out.println("\t\t\t<a href=\"#\" onclick=\"hideAllDetails( '" + rb.getString("link.show.readonly") + "' ); return false;\">" + rb.getString("link.hide.results") + "</a>" );
-    			out.println("\t\t</div>\n\t</td></tr>");
-    		}
+			// unnested citations
+    		displayCitations(out, citations, collection, false, citationCollectionId, title, contentCollectionId);
+
+    		// orlims logos
+    		out.println("<div class='logos'><img src='/sakai-citations-tool/image/sakai/orlims-logo.png' width='82' height='100'><img src='/sakai-citations-tool/image/sakai/bodleian-libraries-logo.png' width='127' height='89'></div>");
```
Comment (buckett) [inline citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java]: I don't think this will want to go back to Sakai.

Comment (nicholaswilson100) [inline citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java]: separate commit or make it configurable?

Comment (buckett) [inline citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java]: Configurable means one less change.

File: citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java [hunk 2]
```diff
@@ -135,7 +131,7 @@ protected void handleExportRequest(HttpServletRequest req, HttpServletResponse r
 			Reference ref, String format, String subtype) 
 			throws EntityNotDefinedException, EntityAccessOverloadException, EntityPermissionException 
 	{
-		if(! ContentHostingService.allowGetResource(req.getParameter("resourceId")))
+		if(! ContentHostingService.allowGetResource(req.getParameter("contentCollectionId")))
```
Comment (buckett) [inline citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java]: This looks similar to a fix Surya was developing is this aiming to fix a permissions issue?

File: citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java [hunk 3]
```diff
@@ -582,5 +364,382 @@ protected void handleViewRequest(HttpServletRequest req, HttpServletResponse res
...
+		}
+	}
+
+	private void displayNestedSections(String title, String citationCollectionId, org.sakaiproject.citation.api.CitationService citationService, CitationCollection collection, CitationCollection fullCollection, PrintWriter out, String contentCollectionId) throws IdUnusedException {
+
+		CitationCollectionOrder nestedCollection = citationService.getNestedCollection(citationCollectionId);
+		int nestedSectionsSize = nestedCollection.getChildren().size();
+
+		out.println("<ol class='serialization vertical h1NestedLevel' style='padding:0;'>");
+		int num = 1;
```
Comment (buckett) [inline citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java]: Isn't this unused? Were you just using it for debugging?

Comment (nicholaswilson100) [inline citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java]: yup i'll remove it!

File: citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java (hunk 2 above)
Comment (nicholaswilson100) [inline citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java]: Yup aiming to fix a permissions issue.  We check if the user has permissions on the content collection before allowing them to export their citations.

Comment (buckett) [inline citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java]: Wasn't this in #11 ?

Why is this the right fix now, you're doing a permission check on one variable, and then exporting the content of another?

Comment (nicholaswilson100) [inline citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java]: Yes I think I remember this from a different Jira. I'll have a look - it's not necessary for this release though , I can take it out .

Comment (buckett) [inline citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java]: I think we should do this in a separate Jira so it gets a good test plan and it much clearer about the bug it's fixing.

**Comment by nicholaswilson100** [comment]
Ok I've taken out the export link so I can have another look at the problem in the original Jira

**Comment by nicholaswilson100** [comment]
I've made the logos configurable

File: citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java [hunk 4]
```diff
@@ -312,254 +309,48 @@ protected void handleViewRequest(HttpServletRequest req, HttpServletResponse res
...
-    							if( !Schema.TITLE.equals(field.getIdentifier()) )
-    							{
-    								out.println("\t\t\t\t<tr>\n\t\t\t\t\t<td class=\"attach\"><strong>" + label + "</strong></td>\n\t\t\t\t\t<td>" + Validator.escapeHtml(value) + "</td>\n\t\t\t\t</tr>");
-    							}
-
-    						}
-    					}
-    				}
+    		// logos
+    		String[] logos = ServerConfigurationService.getStrings("citationslogo");
```
Comment (buckett) [inline citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java]: Most properties use `.` to separate parts so this should be `citations.logo`

Comment (nicholaswilson100) [inline citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java]: ok

**Comment by nicholaswilson100** [comment]
Ok I've added the '.'

**Pull request merged**